src/cli.ts

```diff
@@ -25,35 +39,120 @@ async function main(argv: minimist.ParsedArgs) {
   }
 
   const argIn: string = argv._[0] || argv.input
-  const argOut: string = argv._[1] || argv.output
+  const argOut: string | undefined = argv._[1] || argv.output // the output can be omitted so this can be undefined
+
+  const ISGLOB = isGlob(argIn)
+  const ISDIR = isDir(argIn)
+
+  if ((ISGLOB || ISDIR) && argOut && argOut.includes('.d.ts')) {
+    throw new ReferenceError(
+      `You have specified a single file ${argOut} output for a multi file input ${argIn}. This feature is not yet supported, refer to issue #272 (https://github.com/bcherny/json-schema-to-typescript/issues/272)`
+    )
+  }
 
   try {
-    const schema: JSONSchema4 = JSON.parse(await readInput(argIn))
-    const ts = await compile(schema, argIn, argv as Partial<Options>)
-    await writeOutput(ts, argOut)
+    // Process input as either glob, directory, or single file
+    if (ISGLOB) {
+      await processGlob(argIn, argOut, argv as Partial<Options>)
+    } else if (ISDIR) {
+      await processDir(argIn, argOut, argv as Partial<Options>)
+    } else {
+      await processFile(argIn, argOut, argv as Partial<Options>)
+    }
   } catch (e) {
     console.error(whiteBright.bgRedBright('error'), e)
     process.exit(1)
   }
 }
 
-function readInput(argIn?: string) {
-  if (!argIn) {
-    return new Promise(stdin)
+// check if path is an existing directory
+function isDir(path: string): boolean {
+  return existsSync(path) && lstatSync(path).isDirectory()
+}
+
+async function processGlob(argIn: string, argOut: string | undefined, argv: Partial<Options>) {
+  const files = await glob(argIn) // execute glob pattern match
+
+  if (files.length === 0) {
+    throw ReferenceError(
+      `You passed a glob pattern "${argIn}", but there are no files that match that pattern in ${process.cwd()}`
+    )
   }
-  return readFile(resolve(process.cwd(), argIn), 'utf-8')
+  // create output directory if it does not exist
+  if (argOut && !existsSync(argOut)) {
+    await mkdirp(argOut)
+  }
+
+  return Promise.all(
+    files.map(file => {
+      const outPath = argOut ? `${argOut}/${basename(file, '.json')}.d.ts` : argOut
+      processFile(file, outPath, argv)
+    })
+  )
+}
+
+async function processDir(argIn: string, argOut: string | undefined, argv: Partial<Options>): Promise<void[]> {
+  const files = getPaths(argIn)
+
+  return Promise.all(
+    files.map(file => {
+      if (!argOut) {
+        processFile(file, argOut, argv)
+      } else {
+        /*
+        the following logic determines the out path by comparing the in path to the users specified out path.
+        For example, if input directory MultiSchema looks like:
+        MultiSchema/foo/a.json
+        MultiSchema/bar/fuzz/c.json
+        MultiSchema/bar/d.json
+        And the user wants the outputs to be in MultiSchema/Out, then this code will be able to map the inner directories foo, bar, and fuzz into the intended Out directory.
```

Comment:
This comment is helpful! Mind extending this comment to also include what we'll map these to?

ie.

```
MultiSchema/Out/foo/a.json
MultiSchema/Out/bar/fuzz/c.json
MultiSchema/Out/bar/d.json
```